source/beamline/BioXAS/BioXASBeamline.cpp

```diff
@@ -338,6 +338,19 @@ CLSValves* BioXASBeamline::valves() const
 	return result;
 }
 
+AMXRFDetector *BioXASBeamline::xrfDetector() const
```

Comment:
I agree this might not work in the future. This function should return a default XRF detector. There is another overloaded function in the other beamlines (SXRMB, IDEAS, etc) `AMXRFDetector *xrfDetector(SXRMB::FluorescenceDetectors detectorType) const;`. This will return the XRFDetector with given type (the one configured).